### DISCUSSION TITLE
TransactionScope with multiple connections but no MSDTC.

### DISCUSSION THREAD

**Discussion by ReneElizondo** [MarimerLLC/csla#2913] [Questions] [answered]
Hello everyone,

I have a `DataPortal_Update()` function that is decorated with the `[Transactional(TransactionalTypes.TransactionScope)]` attribute. Everything worked great until I decided to open two different connections Inside the function, when I did that boooom... I got the following error:

**_Network access for Distributed Transaction Manager (MSDTC) has been disabled. Please enable DTC for network access in the security configuration for MSDTC using the Component Services Administrative tool.'_**

The thing is that I am not interested on using MSDTC when I open my second connection, all the second connection does is read some data from a different database so I prefer not to recruit the second connection into the transaction scope. To get around the error, I ended up wrapping the second connection call inside a new transaction scope as follows:

```
using (TransactionScope scope = new TransactionScope(TransactionScopeOption.Suppress))
{
   // Open the new connectino here and execute the query.
    scope.Complete();
}
```

This appears to work fine, but I was wondering if this is asking for trouble and will end up causing issues because of the way the CSLA framework may be handling thigs.

Is this a good workaround? Or are the better approaches to this without complicating things to much.

Thanks.

**Comment by hurcane** [comment, marked as answer]
When you do not want connections to share a TransactionScope, what you have done is a good solution. An alternate technique is to not use TransactionScope at all and manage transactions via SqlTransaction.

**Reply by rockfordlhotka** [reply]
@hurcane is right - if you delegate the creation of the transaction scope to the data portal, it creates/disposes a scope surrounding your data portal operation. If you don't want that behavior, you should probably use manual transactions and manage them yourself.

However, if you don't think it reduces readability, you can do as you suggest - let the data portal manage the "outer" transaction scope, and also create one manually.

Literally all that happens when you use the attribute is that the data portal creates a scope, calls your data portal operation method, and then ensures that the scope is disposed once the operation is complete. Nothing more than you could do yourself - it hopefully just saves some code in the simple/common case.
